### PR TITLE
Add Google Analytics Autotrack

### PR DESCRIPTION
**Pull request by martypowell**
Added a utiltiy to help track clicks to external links

- I forgot to build on my last pr, so some event changes are included in the dist folder

### CODE REVIEW

File: test/utility/bc-google-analytics.spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-undef */
+
+describe('Baltimore County Google Analytics Utility', () => {
+	describe('isExternalLink', () => {
+		it('should return false, if no element is passed in', () => {
+			const actual = bcpl.utility.googleAnalytics.isExternalLink();
+			expect(actual).toEqual(false);
+		});
+
+		it('should return false, the link element contains a link containing the domain bcpl.info', () => {
```
Comment (ascire) [inline test/utility/bc-google-analytics.spec.js]: In the three test cases here where `false` is expected, are these exact links part of the specification, or is the requirement something more like "domain ends with bcpl.info"?

File: test/utility/bc-google-analytics.spec.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			const mockLink = {
+				hostname: 'bcpl.info'
+			};
+			const actual = bcpl.utility.googleAnalytics.isExternalLink(mockLink);
+			expect(actual).toEqual(false);
+		});
+
+		it('should return false, the link element contains a link containing the domain www.bcpl.info', () => {
```
Comment (ascire) [inline test/utility/bc-google-analytics.spec.js]: See previous comment.  Should the requirement here read more like this?
```
it('should return false, the link element contains a link ending with the domain bcpl.info', () => {
```

File: dist/js/apps/events-page/featuredEventsWidgetApp.js [hunk 3]
```diff
@@ -219,7 +219,7 @@
 		requestChunkSize: 10,
 		ageDisclaimer: {
 			message: 'Children under 8 must be accompanied by adult',
```
Comment (jsmith176) [inline dist/js/apps/events-page/featuredEventsWidgetApp.js]: 'Children under 8 must be accompanied by **_an_** adult',

Comment (martypowell) [reply dist/js/apps/events-page/featuredEventsWidgetApp.js]: will run this by dan

File: js/utility/bc-google-analytics.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+// Requires Gtag from Google Analytics, requires includes polyfill
+
+namespacer('bcpl.utility');
+
+bcpl.utility.googleAnalytics = (() => {
+	let gtag;
```
[the remark this answers is not in the document]
Comment (martypowell) [reply js/utility/bc-google-analytics.js]: @ascire yes, added

File: js/utility/bc-google-analytics.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		document.querySelector(document)
+			.addEventListener('click', handleExternalLinkClick);
+	};
+
+	const handleExternalLinkClick = (clickEvent) => {
+		const isTargetAnExternalLinkElm = isExternalLink(clickEvent.target);
+
+		if (isTargetAnExternalLinkElm) {
+			const linkHref = clickEvent.target
+                && Object.prototype.hasOwnProperty.call(clickEvent.target, 'href');
```
[the remark this answers is not in the document]
Comment (martypowell) [reply js/utility/bc-google-analytics.js]: @ascire probably should be a utility, as often as we use it.

File: js/utility/bc-google-analytics.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+// Requires Gtag from Google Analytics, requires includes polyfill
+
+namespacer('bcpl.utility');
+
+bcpl.utility.googleAnalytics = (() => {
+	let gtag;
+	let validHostNames = ['www.bcpl.info', 'bcpl.info', 'catalog.bcpl.lib.md.us', 'www.catalog.bcpl.lib.md.us'];
```
Comment (ascire) [inline js/utility/bc-google-analytics.js]: @martypowell and I discussed in IM, but I want to record somewhere that this hard-coded list of domain names is a maintenance concern.

File: test/utility/bc-google-analytics.spec.js (hunk 1 above)
Comment (martypowell) [reply test/utility/bc-google-analytics.spec.js]: @ascire updated to account for subdomains, please give another look

Comment (ascire) [reply test/utility/bc-google-analytics.spec.js]: @martypowell I don't see the fix yet?

File: js/utility/bc-google-analytics.js [hunk 7]
```diff
@@ -26,10 +29,15 @@ bcpl.utility.googleAnalytics = (() => {
...
-        && linkElm.hostname !== window.location.hostname
-        && !validHostNames.includes(linkElm.hostname));
+		&& hasOwnProperty(linkElm, 'hostname')
+		&& linkElm.hostname
+		&& linkElm.hostname !== window.location.hostname
+		&& !isValidHostName(linkElm.hostname));
+
+	const isValidHostName = (linkHostName) =>
+		!!(validHostNames
+			.filter(validHostName => linkHostName.indexOf(validHostName) > -1)
```
[the remark this answers is not in the document]
Comment (martypowell) [reply js/utility/bc-google-analytics.js]: @ascire updated

**Comment by martypowell** [comment]
@ascire ready for another look.

**Pull request merged**